pkgs/development/tools/sourcetrail/python.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv, lib, fetchFromGitHub, wrapPython, python, jedi, parso, cmake, swig }:
+
+stdenv.mkDerivation rec {
+  pname = "SourcetrailPythonIndexer";
+  # FIXME: use release version when version > v1_db25_p4 is out
+  #
+  # The latest commits includes SourcetrailDB as a submodule, but the latest
+  # release still doesn't have this change. Since it's annoying to have to
+  # do things differently just for the current release, we just use the latest
+  # commit for now. It shouldn't be much of a problem since the only difference
+  # is just maintenance stuff.
+  version = "0d1246be53b37639427fe9959f748dd0fc429c8f";
+
+  src = fetchFromGitHub {
+    owner = "CoatiSoftware";
+    repo = pname;
+    rev = version;
+    sha256 = "05hlpd3am029pv1wi6mys3q0ggp64axmg8bdf1fabl9cl9jffscq";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ wrapPython cmake swig ];
+  buildInputs = [ python ];
+  pythonPath = [ jedi parso ];
+
+  dontUseCmakeConfigure = true;
+  cmakeFlags = [
+    "-DBUILD_BINDINGS_PYTHON=1"
+    "-DPYTHON_VERSION=${lib.versions.majorMinor python.version}"
+  ];
+
+  buildPhase = ''
+    pushd SourcetrailDB
+    cmake -Bbuild $cmakeFlags .
+    pushd build
+    make -j $NIX_BUILD_CORES
+    popd
+    popd
+  '';
+
+  installPhase = ''
+    shopt -s extglob
+    mkdir -p $out/{bin,libexec}
+
+    cp !(run).py $out/libexec
```

Comment:
Is this right` `!(run).py`?

Reply:
That matches every python script except `run.py`. It works because I've enabled extended globbing in L42. https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Pattern-Matching

Reply:
Oh. Bash is such a huge bag of tricks, impossible to know them all. Probably also not advisable to use them all, but if it makes things simpler in this case why not. Maybe a comment would be helpful though, since that is not very commonly used.

Reply:
I added a brief comment.